MapboxMobileEvents/MMEAPIClient.m

```diff
@@ -181,6 +213,49 @@ - (NSURLRequest *)requestForEvents:(NSArray *)events {
     return [request copy];
 }
 
+- (NSString *)mimeTypeForPath:(NSString *)path {
+    CFStringRef extension = (__bridge CFStringRef)[path pathExtension];
+    CFStringRef UTI = UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, extension, NULL);
+    if (UTI == NULL) {
+        return nil;
+    }
+    
+    NSString *mimetype = CFBridgingRelease(UTTypeCopyPreferredTagWithClass(UTI, kUTTagClassMIMEType));
+    if (mimetype == NULL) {
+        return nil;
+    }
+    
+    CFRelease(UTI);
+    
+    return mimetype;
+}
+
+- (NSData *)createBodyWithBoundary:(NSString *)boundary metadata:(NSArray *)metadata filePaths:(NSArray *)filePaths {
+    NSMutableData *httpBody = [NSMutableData data];
+    
+    [httpBody appendData:[[NSString stringWithFormat:@"--%@\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];
+    [httpBody appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"attachments\"\r\n"] dataUsingEncoding:NSUTF8StringEncoding]];
+    [httpBody appendData:[[NSString stringWithFormat:@"Content-Type: application/json\r\n\r\n"] dataUsingEncoding:NSUTF8StringEncoding]];
+    [httpBody appendData:[NSJSONSerialization dataWithJSONObject:metadata options:0 error:nil]];
+    [httpBody appendData:[[NSString stringWithFormat:@"\r\n\r\n"] dataUsingEncoding:NSUTF8StringEncoding]];
+    
+    for (NSString *path in filePaths) {
+        NSString *filename  = [path lastPathComponent];
+        NSData   *data      = [NSData dataWithContentsOfFile:path];
+        NSString *mimetype  = [self mimeTypeForPath:path];
+        
+        [httpBody appendData:[[NSString stringWithFormat:@"--%@\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];
+        [httpBody appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"file\"; filename=\"%@\"\r\n", filename] dataUsingEncoding:NSUTF8StringEncoding]];
+        [httpBody appendData:[[NSString stringWithFormat:@"Content-Type: %@\r\n\r\n", mimetype] dataUsingEncoding:NSUTF8StringEncoding]];
```

Comment:
What happens here if `mimetype` is `nil`? I'd be tempted to wrap the appendData line with an `if (mimetype != nil)`.

Reply:
It's actually accepted server-side without a mimetype at the moment.